### PR TITLE
update docs for sdkMiddlewareAuth

### PR DESCRIPTION
**Pull request by adnasa**
- remove `user` parameter since it is not needed
- correct syntax error
- because the syntax error is not corrected, consequently prettier kicks in.

commit-by-commit review is possible

### CODE REVIEW

File: docs/sdk/api/sdkMiddlewareAuth.md [hunk 1]
```diff
@@ -40,15 +42,8 @@ const client = createClient({
       credentials: {
         clientId: '123',
         clientSecret: 'secret',
-        user: {
```
Comment (gnerkus) [inline docs/sdk/api/sdkMiddlewareAuth.md]: Is this because we don't have any user authentication?

Comment (gnerkus) [reply docs/sdk/api/sdkMiddlewareAuth.md]: NVM.

I noticed it's not required for the 'client credential flow'.

File: docs/sdk/api/sdkMiddlewareAuth.md [hunk 2]
```diff
@@ -60,12 +55,11 @@ Creates a [middleware](/sdk/Glossary.md#middleware) to handle authentication for
...
 
-1. `host` *(String)*: the host of the OAuth API service
-2. `projectKey` *(String)*: the key of the project to assign the default scope to
-   - The user field is an object containining `username` and `password`. [Sample below](#usage-example)
-3. `credentials` *(Object)*: the client credentials for authentication (`clientId`, `clientSecret`)
-4. `scopes` *(Array)*: a list of [scopes](http://dev.commercetools.com/http-api-authorization.html#scopes) to assign to the OAuth token. _No default scope is sent_
-
+1. `host` _(String)_: the host of the OAuth API service
+2. `projectKey` _(String)_: the key of the project to assign the default scope to
+   * The user field is an object containining `username` and `password`. [Sample below](#usage-example)
```
Comment (wizzy25) [inline docs/sdk/api/sdkMiddlewareAuth.md]: Thanks for this @adnasa 💯 🥇 
While we're at this, add the `user` on the next line, and move this line below it? Also, the `Sample below` link references the example from `clientCredentialsFlow` which you have refactored away. Could you fix this to link to the proper usage example? Thanks.

Comment (adnasa) [reply docs/sdk/api/sdkMiddlewareAuth.md]: > Also, the Sample below link references the example from clientCredentialsFlow which you have refactored away

I think you're looking at the wrong line. I refactored away users from `createAuthMiddlewareForClientCredentialsFlow` where as this link is referring to the example in `createAuthMiddlewareForPasswordFlow`

Comment (wizzy25) [reply docs/sdk/api/sdkMiddlewareAuth.md]: The (#usage-example) should be (#usage-example-1) as they point to different usage examples

File: docs/sdk/api/sdkMiddlewareAuth.md [hunk 3]
```diff
@@ -60,12 +55,11 @@ Creates a [middleware](/sdk/Glossary.md#middleware) to handle authentication for
...
-1. `host` *(String)*: the host of the OAuth API service
-2. `projectKey` *(String)*: the key of the project to assign the default scope to
-   - The user field is an object containining `username` and `password`. [Sample below](#usage-example)
-3. `credentials` *(Object)*: the client credentials for authentication (`clientId`, `clientSecret`)
-4. `scopes` *(Array)*: a list of [scopes](http://dev.commercetools.com/http-api-authorization.html#scopes) to assign to the OAuth token. _No default scope is sent_
-
+1. `host` _(String)_: the host of the OAuth API service
+2. `projectKey` _(String)_: the key of the project to assign the default scope to
+3. `credentials` _(Object)_: the client credentials for authentication (`clientId`, `clientSecret`, `user`)
+  * The `user` field is an object containing `username` and `password`. [Sample below](#usage-example)
```
Comment (wizzy25) [inline docs/sdk/api/sdkMiddlewareAuth.md]: (#usage-example) should be (#usage-example-1) here

Comment (adnasa) [reply docs/sdk/api/sdkMiddlewareAuth.md]: 2489207 (via `git commit --amend`

**Pull request merged**